### PR TITLE
refactor: create MapWithLocationsAndPath component

### PR DESCRIPTION
**Pull request by NoamGaash**
# Description
refactored some components of the `SingleLineMap` page to make it easier to deal with
in the future, and prepare its map component for use in the line profile page

## screenshots
irrelevant

### CODE REVIEW

**Comment by amabelleS** [comment]
I have not reviewed all the changes, To do it properly I will need more time to understand them. 
But I see the build has failed:
image

**Comment by NoamGaash** [comment]
@amabelleS you're right, solved :pray: :smile:

File: src/pages/components/Widget.stories.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import type { Meta, StoryObj } from '@storybook/react'
-import { BusToolTip } from './MapLayers/BusToolTip'
+import { BusToolTip } from './map-related/MapLayers/BusToolTip'
```
Comment (shootermv) [inline src/pages/components/Widget.stories.tsx:2]: why this import needed here?

Comment (NoamGaash) [reply src/pages/components/Widget.stories.tsx:2]: it's not. it should be removed. it's not related to this PR. it should be fixed

Comment (NoamGaash) [reply src/pages/components/Widget.stories.tsx:2]: you're the one to blame actually :sweat_smile: 
image

Comment (shootermv) [reply src/pages/components/Widget.stories.tsx:2]: Shame on me!

Comment (NoamGaash) [reply src/pages/components/Widget.stories.tsx:2]: no shame in [blame](https://git-scm.com/docs/git-blame)! :wink:

**Comment by NoamGaash** [comment]
@shootermv @amabelleS

**Pull request merged**